relay-server/src/envelope.rs

```diff
@@ -739,6 +745,17 @@ impl Item {
         self.headers.source_quantities = Some(source_quantities);
     }
 
+    /// Returns if the payload's replay items should be combined into one kafka message.
+    #[cfg_attr(not(feature = "processing"), allow(dead_code))]
```

Comment:
Why not just

```suggestion
    #[cfg(processing)]
```

?